### PR TITLE
Simplify `enforce_signatures`

### LINKED ISSUE #49: Inline visibility modifiers cause false positives in Sorbet/EnforceSignatures
To repro, create `a.rb` with:
```ruby
# typed: strict

class A
  sig { void }
  private def one
    1
  end
end
```
Expected result: no `Sorbet/EnforceSignatures` violation
Actual result:
```sh
$ rubocop --only Sorbet/EnforceSignatures a.rb
Inspecting 1 file
C

Offenses:

a.rb:5:11: C: [Correctable] Sorbet/EnforceSignatures: Each method is required to have a signature.
  private def one ...
          ^^^^^^^

1 file inspected, 1 offense detected, 1 offense auto-correctable
```

The autocorrect here is also a bit busted:
```
$ rubocop --only Sorbet/EnforceSignatures -a a.rb
Inspecting 1 file
C

Offenses:

a.rb:5:11: C: [Corrected] Sorbet/EnforceSignatures: Each method is required to have a signature.
  private def one ...
          ^^^^^^^
a.rb:6:11: C: [Corrected] Sorbet/EnforceSignatures: Each method is required to have a signature.
          def one ...
          ^^^^^^^

1 file inspected, 2 offenses detected, 2 offenses corrected
$ cat a.rb
# typed: strict

class A
  sig { void }
  private sig { returns(T.untyped) }
          sig { returns(T.untyped) }
          def one
    1
  end
end

### PR DESCRIPTION
**Pull request by Morriar**
Just look if there is a dangling signature in the current scope when we encounter a new def or attribute definition.

Fixes #49.

### CODE REVIEW

**Review by KaanOzkan — APPROVED** [review]
LGTM pending the rubocop offense.

**Review by RyanBrushett — APPROVED** [review]

File: lib/rubocop/cop/sorbet/signatures/enforce_signatures.rb [hunk 1]
```diff
@@ -63,22 +71,23 @@ def autocorrect(node)
           end
         end
 
+        def scope(node)
+          return nil unless node.parent
+          return node.parent if [:begin, :class, :module].include?(node.parent.type)
```
Comment (paracycle) [inline lib/rubocop/cop/sorbet/signatures/enforce_signatures.rb:76]: What about blocks?

File: spec/rubocop/cop/sorbet/signatures/enforce_signatures_spec.rb [hunk 2]
```diff
@@ -45,6 +45,48 @@ def foo; end
...
+          sig { void }
+          def foo; end
+        end
+
+        sig { void }
+        sig { void }
+        def foo; end
+      RUBY
+    end
+    it 'handles scopes correctly' do
```
Comment (paracycle) [inline spec/rubocop/cop/sorbet/signatures/enforce_signatures_spec.rb:64]: Should we test sigs in blocks as well?

File: lib/rubocop/cop/sorbet/signatures/enforce_signatures.rb (hunk 1 above)
Comment (Morriar) [reply lib/rubocop/cop/sorbet/signatures/enforce_signatures.rb]: Good catch, I was storing the sig in the block's parent scope!

File: spec/rubocop/cop/sorbet/signatures/enforce_signatures_spec.rb (hunk 2 above)
Comment (Morriar) [reply spec/rubocop/cop/sorbet/signatures/enforce_signatures_spec.rb:65]: Added a test

**Review by paracycle — APPROVED** [review]

**Pull request merged**